### DISCUSSION TITLE
TextBox with revealPasswordButton should be made read-only

### DISCUSSION THREAD

**Discussion by apfohl** [AvaloniaUI/Avalonia#6882] [Ideas]
Hi,

I came across a usage of the `Classes="revealPasswordButton"` feature for the `TextBox` control. I assume the suggested use case right now is to have the TextBox show the password if the users wants to unlock something, so she can check if there are any typos. In my case I want to show password in the case of a password manager. I implemented a view which shows the hidden password and reveals it on the click of a button. Preferably with the `revealPasswordButton` feature. As it doesn't introduce new code.

But right now the selector inside the TextBox style look like follows:

    TextBox.revealPasswordButton[AcceptsReturn=False][IsReadOnly=False]:not(TextBox:empty)

In my use case the user shall not be able to edit the password. If I set the `IsReadOnly` property to `true` the style is not applied.

What do you think about this. Is it worth changing the selector to accept also read-only TextBoxes?

**Comment by gambergeurr** [comment]
hey please make this a thing
